katdal/lazy_indexer.py

```diff
@@ -330,3 +330,39 @@ def dtype(self):
         """Type of data array after transformation, i.e. `self[:].dtype`."""
         return reduce(lambda dtype, transform: transform.dtype if transform.dtype is not None else dtype,
                       self.transforms, self._initial_dtype)
+
+
+class DaskLazyIndexer(object):
+    """Turn a dask Array into a LazyIndexer by computing it upon indexing."""
+    def __init__(self, dataset, keep=(), transforms=None):
+        self.name = getattr(dataset, 'name', '')
+        try:
+            dataset = dataset[keep]
+        except NotImplementedError:
+            # Dask does not like multiple boolean indices: go one dim at a time
+            for dim, keep_per_dim in enumerate(keep):
+                index = dataset.ndim * [slice(None)]
+                index[dim] = keep_per_dim
+                dataset = dataset[tuple(index)]
+        self.dataset = dataset
+        self.transforms = [] if transforms is None else transforms
```

Comment:
Where do these get applied?

Reply:
They don't - I left them for compatibility with the old indexer. I'll just force it to be empty.

Reply:
If they're not applied then it's not really compatible. Maybe remove the constructor argument so that if someone tries to actually pass a transform they get an error early. Leaving the attribute (as an empty list) would help with compatibility.

Reply:
Fixed.